x-pack/plugins/threat_intelligence/public/containers/integrations_guard/integrations_guard.tsx

```diff
@@ -7,26 +7,75 @@
 
 import { EuiLoadingLogo } from '@elastic/eui';
 import React, { FC } from 'react';
+import { HttpSetup } from '@kbn/core/public';
+import { useRequest, UseRequestConfig } from '@kbn/es-ui-shared-plugin/public';
 import { EmptyPage } from '../../modules/empty_page';
 import { useIndicatorsTotalCount } from '../../modules/indicators';
 import { SecuritySolutionPluginTemplateWrapper } from '../security_solution_plugin_template_wrapper';
 
+type IntegrationInstallStatus = 'installed' | 'installing' | 'install_failed';
+
+let httpClient: HttpSetup;
+
+const PACKAGES_CONFIG: UseRequestConfig = {
+  path: '/api/fleet/epm/packages',
+  method: 'get',
+};
+
+export const THREAT_INTELLIGENCE_CATEGORY = 'threat_intel';
+
+export const THREAT_INTELLIGENCE_UTILITIES = 'ti_util';
+
+export const installationStatuses = {
+  Installed: 'installed',
+  Installing: 'installing',
+  InstallFailed: 'install_failed',
+  NotInstalled: 'not_installed',
+};
+
+interface Integration {
+  categories: string[];
+  id: string;
+  status: IntegrationInstallStatus;
+}
+
+export const setHttpClient = (client: HttpSetup) => {
+  httpClient = client;
+};
+
 /**
- * Renders children only if TI integrations are enabled
+ * Renders children only if TI integrations are enabled and indicators are received
  */
 export const IntegrationsGuard: FC = ({ children }) => {
-  const { count: indicatorsTotalCount, isLoading: isIndicatorsTotalCountLoading } =
+  const { isLoading: indicatorsTotalCountLoading, count: indicatorsTotalCount } =
     useIndicatorsTotalCount();
 
-  if (isIndicatorsTotalCountLoading) {
+  const { isLoading: packagesLoading, data: integrations } = useRequest(
+    httpClient,
+    PACKAGES_CONFIG
+  );
+
+  if (packagesLoading && indicatorsTotalCountLoading) {
```

Comment:
I think `||` will be better here to avoid flickering the page when one request is resolved and the other one is not. For example, the request for integrations resolves first and there are no integrations so we show the empty page and then the count request resolves and returns a non-zero count, so we show the indicator page right after. Btw these cases might make sense to test (the different order of resolution with different results) so we make sure users don't end up in a weird state

Reply:
using `&&` guaranteed to show something as soon as the `indicators` call would finish, so I felt like it was a good solution. I didn't see a flickering during my testing but I also didn't try to run with a limited network connection.
I'll do more testing, thanks!

Reply:
I'm now making the request using `react-query` instead of `useRequest` and I feel like I can now use `||` instead of `&&`.
I've tested with the endpoint returning 404 and the indicators page shows up as expected.
I also went with a stale time of 2 sec but happy to change it to anything else. From my brief testing, the call comes back in hundreds of milliseconds...

Reply:
my concerns are more about edge cases and occasional problems with networks. I'm absolutely sure that in most cases the response will be fast, but from my experience, we shouldn't rely on that fact - for some users the fleet endpoint might sometime respond after much longer for who knows what reason. It's better to be mindful of these known unknowns